src/nativeMain/kotlin/com/autodesk/coroutineworker/WorkerPool.kt

```diff
@@ -69,42 +59,19 @@ internal class WorkerPool(private val numWorkers: Int) {
         val worker = nextWorker()
 
         // prepare the block to update state, when the worker is finished
-        val workerCompleteBlock = {
+        val workerCompleteBlock: () -> Unit = {
             nextWorkerLock.withLock {
                 worker.numBlocksQueue.decrement()
             }
-        }.freeze()
-        val future = worker.worker.execute(
-            TransferMode.SAFE,
-            { Pair(work.freeze(), workerCompleteBlock) }) { (work, complete) ->
+        }
+        val workerOperation: () -> Unit = {
             try {
                 work()
             } finally {
-                complete()
+                workerCompleteBlock()
             }
-        }
-        cleanupFinishedFuturesAndAdd(future)
-    }
-
-    private fun cleanupFinishedFuturesAndAdd(futureToAdd: Future<*>) {
```

Comment:
Can you explain why we no longer need any of this futures related stuff?

Reply:
The new executeAfter API doesn't return one